tokio/src/runtime/basic_scheduler.rs

```diff
@@ -298,10 +302,16 @@ where
         }
 
         // Wait until all tasks have been released.
-        while unsafe { self.scheduler.queues.has_tasks_remaining() } {
-            self.local.park.park().ok().expect("park failed");
+        loop {
             unsafe {
                 self.scheduler.queues.drain_pending_drop();
+                self.scheduler.queues.drain_queues();
+
+                if !self.scheduler.queues.has_tasks_remaining() {
+                    break;
+                }
+
+                self.local.park.park().ok().expect("park failed");
```

Comment:
Nit/TIOLI: this is safe to call, right? Maybe it should be moved out of the `unsafe` block?